nixos/lib/testing-python.nix

```
@@ -159,7 +171,10 @@ rec {
 
             echo -n "$testScript" > $out/test-script
             ${lib.optionalString (!skipLint) ''
-              ${python3Packages.black}/bin/black --check --diff $out/test-script
+              PYFLAKES_BUILTINS="$(
+                echo -n ${lib.escapeShellArg (lib.concatStringsSep "," nodeNames)},
```

```suggestion
                echo -n ${lib.escapeShellArg (lib.concatStringsSep "," nodeHostNames)},
```

Subtle but important for amazon-init-shell and cloud-init.

---

Eeek, so this essentially means that you can create colliding nodes?

---

It's not great. Having two namespace is nice though if you want hostnames with `.`'s in them, but that's not even supported now. Something for later.